### PR TITLE
Enable github actions ci

### PR DESCRIPTION
**Pull request by CAD97**

### CODE REVIEW

**Comment by CAD97** [comment]
This was meant to be a self-pr for testing how did I mess up this badly GitHub UI please

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+          components: rustfmt, rust-src
+      - name: Compile
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --tests
```
Comment (matklad) [inline .github/workflows/ci.yaml]: This should be `test —no-run`, the build is only for integration tests, which we don’t ise

**Comment by CAD97** [comment]
Here's what I got with an attempt at literally translating just the CI part of travis.yml: https://github.com/CAD97/libsyntax2/runs/306447310

I don't know how to use xvfb but the cache seems to be working successfully. Note that IIUC, you version the cache key manually, so we _might_ be able to remove the `Prepare build directory for cache` step and lose nothing (other than having a larger cache file).

The Rust Tests Workflow ran in 7m 26s, and it looks like Travis runs the equivalent job in about 9.5 minutes.

**Comment by matklad** [comment]
@CAD97 LGTM! Yeah, let's just switch to GitHub actions, to keep the number of external services as small as possible. 

Could you send what you have as a PR, and I'll figure out the rest (permissions for doc publishing and bors)?

**Pull request closed without merging**